component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/dao/JWTStorageManager.java

```diff
@@ -94,44 +104,85 @@ public JWTEntry getJwtFromDB(String jti) throws OAuthClientAuthnException {
         try {
             prepStmt = dbConnection.prepareStatement(Constants.SQLQueries.GET_JWT);
             prepStmt.setString(1, jti);
+            prepStmt.setInt(2, tenantId);
             rs = prepStmt.executeQuery();
             if (rs.next()) {
                 long exp = rs.getTime(1, Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
                 long created = rs.getTime(2, Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC))).getTime();
-                jwtEntry = new JWTEntry(exp, created);
+                jwtEntry = new JWTEntry(exp, created, tenantId);
             }
         } catch (SQLException e) {
             if (log.isDebugEnabled()) {
                 log.debug("Error when retrieving the JWT ID: " + jti, e);
             }
             throw new OAuthClientAuthnException("Error occurred while validating the JTI: " + jti + " of the " +
-                                                "assertion.", OAuth2ErrorCodes.INVALID_REQUEST);
+                    "assertion.", OAuth2ErrorCodes.INVALID_REQUEST);
         } finally {
             IdentityDatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
         }
         return jwtEntry;
     }
 
+    /**
+     * To get a list of persisted JWTs for a given JTI.
+     *
+     * @param jti JTI.
+     * @return List of JWTEntries.
+     * @throws OAuthClientAuthnException OAuthClientAuthnException thrown with Invalid Request error code.
+     */
+    public List<JWTEntry> getJwtsFromDB(String jti, int tenantId) throws OAuthClientAuthnException {
+
+        List<JWTEntry> JWTEntries = new ArrayList<>();
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+        try {
+            prepStmt = dbConnection.prepareStatement(Constants.SQLQueries.GET_JWT_DETAIL);
+            prepStmt.setString(1, jti);
+            prepStmt.setInt(2, tenantId);
+            prepStmt.setInt(3, DEFAULT_TENANT_ID);
```

Comment:
Seems this parameter is not used in the SQL Query. Wondering why we need 'DEFAULT_TENANT_ID' ?